doc/design/draft/subscription.md

```diff
@@ -0,0 +1,180 @@
+# Subscription
+
+## Summary
+
+In this document we analyse how many and what kind of chunks will be provided
+to a subscriber on subscription. The answer depends on the number of publishers,
+how many samples were send in the past and how large the requested _HistorySize_
+is.
+
+The goal is to craft and implement a design which is unable to deadlock,
+massively slow down (caused by deadlock discovery) or crash RouDi.
+
+## Terminology
+
+- **ChunkHistory** the last N samples which the publishers of a service have
+   published
+- **HistorySize** the number of samples the subscriber will receive at most on
+   subscription
+- **DeliveryTime** the point in time where the publisher writes the samples in
+   the _ChunkHistory_. This is the first task right before the actual delivery to
+   the subscribers starts.
+
+## Requirements
+
+### General
+
+- RouDi is allowed to block any arbitrary user process.
+- RouDi is allowed to block threads owned by RouDi.
+- No process is allowed to block any thread in RouDi via logic or lock.
+
+Since RouDi is our central daemon and the whole system is corrupted when it fails
+we can assume for now that RouDi with its inner logic never fails. But we have to
+ensure that RouDi is not corrupted by any external instance like a publisher
+application which holds a lock and dies which would lead to a deadlock in RouDi.
+
+### Subscribing without concurrent send
+
+Let's assume a subscriber S would like to subscribe to a service which has two
+publishers P1 and P2. Furthermore, we assume that P1 and P2 have send the
+following samples:
+
+| Publisher | Timepoint | Sample | | Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|-|:---------:|:---------:|:------:|
+| P1        | 1         | A      | | P2        | 20        | F      |
+| P1        | 11        | B      | | P2        | 22        | G      |
+| P1        | 21        | C      | | P2        | 24        | H      |
+| P1        | 31        | D      | | P2        | 40        | I      |
+| P1        | 41        | E      | | P2        | 42        | J      |
+
+Let's assume the subscriber subscribes to the service.
+
+#### Subscription with _HistorySize_ 0
+
+The subscriber will not be provided with samples from the history.
+
+#### Subscription with _HistorySize_ 1
+
+The subscriber cache contains the one sample ordered after _DeliveryTime_.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+
+#### Subscription with _HistorySize_ 6
+
+The subscriber cache contains the six samples ordered after _DeliveryTime_.
```

Comment:
We cannot reason about order by delivery time unless the clocks are synchronized. Essentially the MPMC queue used to deliver the data will generate (define) the ordering, but it is not necessarily the one defined by time points. 

The only guarantee we have is that it respects the partial orders of both individual publishers.

Reply:
See comment: https://github.com/eclipse-iceoryx/iceoryx/pull/780/files#r654320386

Can be solved very easily with an incrementing number stored in the service description inside of the shared memory. We do not need clock synchronization.